diff -rupN u-boot-2010.03.orig/common/image.c u-boot-2010.03/common/image.c
--- u-boot-2010.03.orig/common/image.c	2010-03-31 23:54:39.000000000 +0200
+++ u-boot-2010.03/common/image.c	2010-04-12 15:42:15.911858000 +0200
@@ -941,7 +941,7 @@ int boot_get_ramdisk (int argc, char *ar
 			return 1;
 		}

-#if defined(CONFIG_B2) || defined(CONFIG_EVB4510) || defined(CONFIG_ARMADILLO)
+#if defined(CONFIG_B2) || defined(CONFIG_EVB4510) || defined(CONFIG_ARMADILLO) || defined(CONFIG_VERSATILE)
 		/*
 		 * We need to copy the ramdisk to SRAM to let Linux boot
 		 */
diff -rupN u-boot-2010.03.orig/include/configs/versatile.h u-boot-2010.03/include/configs/versatile.h
--- u-boot-2010.03.orig/include/configs/versatile.h	2010-03-31 23:54:39.000000000 +0200
+++ u-boot-2010.03/include/configs/versatile.h	2010-04-12 15:43:01.514733000 +0200
@@ -124,8 +124,11 @@
 #define CONFIG_BOOTP_SUBNETMASK

 #define CONFIG_BOOTDELAY	2
-#define CONFIG_BOOTARGS		"root=/dev/nfs mem=128M ip=dhcp "\
-				"netdev=25,0,0xf1010000,0xf1010010,eth0"
+/*#define CONFIG_BOOTARGS		"root=/dev/nfs mem=128M ip=dhcp "\
+				"netdev=25,0,0xf1010000,0xf1010010,eth0"*/
+#define CONFIG_BOOTARGS		"root=/dev/ram mem=128M rdinit=/sbin/init"
+#define CONFIG_BOOTCOMMAND   "bootm 0x210000 0x410000"
+#define CONFIG_INITRD_TAG   1

 /*
  * Static configuration when assigning fixed address
